packages/dashboard/src/components/iot-dashboard/getDashboardPosition.ts

```diff
@@ -2,9 +2,30 @@ import { Position } from '../../types';
 
 export const DASHBOARD_CONTAINER_ID = 'container';
 
-export const getDashboardPosition = (event: MouseEvent): Position => {
-  let totalOffsetX = event.offsetX;
-  let totalOffsetY = event.offsetY;
+const getOffsets = (event: MouseEvent | TouchEvent | PointerEvent) => {
+  if ((event as TouchEvent).touches) {
```

Comment:
is there a way to utilize narrowing (https://www.typescriptlang.org/docs/handbook/2/narrowing.html) to achieve this without asserting `as`?

Reply:
yup! good callout.